plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/model/VFSDirectory.java

```diff
@@ -96,4 +99,26 @@ public static VFSDirectory create( String parent, FileObject fileObject, String
     }
     return vfsDirectory;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash( getProvider(), getConnection(), getPath() );
+  }
+
+  @Override public boolean equals( Object obj ) {
+    // If the object is compared with itself then return true
+    if ( obj == this ) {
+      return true;
+    }
+
+    if ( !( obj instanceof VFSFile ) ) {
+      return false;
+    }
+
+    VFSDirectory compare = (VFSDirectory) obj;
+    return compare.getProvider().equals( getProvider() )
```

Comment:
can't you use this for cleaner code:
`Objects.equals(compare.getProvider(), getProvider()) && Objects.equals(compare.getConnection(), getConnection()) && ...`